sys/reboot/src/log_reboot.c

```diff
@@ -383,6 +384,50 @@ reboot_conf_export(void (*func)(char *name, char *val),
     return 0;
 }
 
+void
+log_reboot_reason_str(enum hal_reset_reason reason, char *buf)
+{
+    char str_reason[MYNEWT_VAL(REBOOT_LOG_REBOOT_REASON_SIZE)];
+
+    if (reason >= HAL_RESET_OTHER) {
+        snprintf(str_reason,MYNEWT_VAL(REBOOT_LOG_REBOOT_REASON_SIZE),"OTHER: 0x%X",reason - HAL_RESET_OTHER);
+        strcpy(buf, str_reason);
+        return;
+    }
+
+    switch (reason) {
+    case HAL_RESET_POR:
+        snprintf(str_reason,MYNEWT_VAL(REBOOT_LOG_REBOOT_REASON_SIZE),"HARD");
```

Comment:
you can just return e.g. `"HARD"` for plain strings
for values that have to be converted to string just do `snprintf` into static buffer of fixed size which is large enough to hold that string and return pointer to that buffer - it's safe as long as it is 'static`